R/restrict_derivation.R

```diff
@@ -78,11 +75,8 @@ restrict_derivation <- function(dataset,
                                 filter) {
   # Check input
   assert_data_frame(dataset)
-  assert_function(derivation, params = c("dataset"))
+  assert_function(derivation)
   assert_s3_class(args, "params", optional = TRUE)
-  if (!is.null(args)) {
-    assert_function(derivation, names(args))
-  }
```

Comment:
I think we do not need the `if` condition but we should keep the `assert_function()` call. Then we could remove the first call of `assert_function()` in line 78.

Reply:
fixed